File: samples/petstore/php/tests/BaseApiTest.php
<?php

// Unit tests for Swagger Petsore sample app PHP client

// Run all tests:
//
//     phpunit tests/PetApiTest.php
//     phpunit tests/StoreApiTest.php
//     phpunit tests/UserApiTest.php

// If you require PHPUnit to be installed, first get PEAR:
//
// $ wget http://pear.php.net/go-pear.phar
// $ php -d detect_unicode=0 go-pear.phar
//
// Then install PHPUnit:
//
// $ pear config-set auto_discover
// $ pear install pear.phpunit.de/PHPUnit

// If you want to see all the JSON, make sure you append the fake api-key:
// http://petstore.swagger.wordnik.com/api/pet.json?api_key=special-key

// You need to set this for your local time zone if not already set in
// global config.
date_default_timezone_set("America/Los_Angeles");

require_once 'petstore/Swagger.php';

class BaseApiTest extends PHPUnit_Framework_TestCase {

  public function setUp() {
    $this->apiKey = "special-key";
    $this->apiUrl = "http://petstore.swagger.wordnik.com/api";
    $this->username = "test";
    $this->password = "test";
    $this->client = new APIClient($this->apiKey, $this->apiUrl);
    $this->petApi = new PetApi($this->client);
    $this->storeApi = new StoreApi($this->client);
    $this->userApi = new UserApi($this->client);
  }

  public function tearDown() {
      unset($this->client);
  }

}


?>

File: samples/client/petstore/python3/tests/BaseApiTest.py
#!/usr/bin/env python
"""Unit tests for SwaggerPython Petstore sample app API client.

Run all tests:

    python BaseApiTest.py

"""

import sys
import os
import unittest

sys.path = ['./'] + sys.path
from petstore import *


class BaseApiTest(unittest.TestCase):

    def setUp(self):
        self.apiKey = 'special-key'
        self.apiUrl = 'http://petstore.swagger.wordnik.com/api'
        self.username = 'test'
        self.password = 'test'

        client = swagger.ApiClient(self.apiKey, self.apiUrl)
        self.petApi = PetApi.PetApi(client)
        self.storeApi = StoreApi.StoreApi(client)
        self.userApi = UserApi.UserApi(client)

if __name__ == "__main__":

    from PetApiTest import PetApiTest
    from StoreApiTest import StoreApiTest
    from UserApiTest import UserApiTest

    unittest.main()
